.travis.yml

```diff
@@ -1,51 +1,71 @@
 language: php
 
 php:
-  - 5.3
-  - 5.4
-  - 5.5
   - 5.6
+  - 7.0
+  - 7.1
+  - 7.2
 
 env:
-  - TYPO3_BRANCH=master COVERAGE=0
-  - TYPO3_BRANCH=TYPO3_6-2 COVERAGE=0
+  - TYPO3_VERSION=^7.6 COVERAGE=0 FUNCTIONAL=0
 
 matrix:
-  exclude:
-    - env: TYPO3_BRANCH=master COVERAGE=0
-      php: 5.3
-    - env: TYPO3_BRANCH=master COVERAGE=0
-      php: 5.4
-    - env: TYPO3_BRANCH=master COVERAGE=0
-      php: 5.5
-    - env: TYPO3_BRANCH=master COVERAGE=0
-      php: 5.6
   include:
-    - php: 5.5
-      env: TYPO3_BRANCH=TYPO3_6-2 COVERAGE=1
+    - env: TYPO3_VERSION=^7.6 COVERAGE=1 FUNCTIONAL=1
+      php: 7.1
+  allow_failures:
+    - php: 7.2
 
-before_script:
-  - sudo apt-get install parallel
+sudo: false
+
+addons:
+  apt:
+    packages:
+      - parallel
+
+cache:
+  directories:
+    - $HOME/.composer/cache
+
+before_install:
   - composer self-update
-  - cd ..
-  - git clone --single-branch --branch $TYPO3_BRANCH --depth 1 https://github.com/TYPO3/TYPO3.CMS.git typo3_core
-  - mv typo3_core/* .
-  - mkdir -p typo3conf/ext
-  - mv happy_feet typo3conf/ext/
-  - composer install
+  - composer --version
+
+before_script:
+  # Installs TYPO3
+  - composer require typo3/cms=$TYPO3_VERSION
+  # Restore composer.json
+  - git checkout composer.json
+  - export "TYPO3_PATH_WEB"=$PWD/.Build/Web;
+  # Locating UnitTests.xml
+  - export "UNIT_XML"='.Build/vendor/nimut/testing-framework/res/Configuration/UnitTests.xml'
+  # Location FunctionalTests.xml
+  - export "FUNCTIONAL_XML"='.Build/vendor/nimut/testing-framework/res/Configuration/FunctionalTests.xml'
+  # Symlink TYPO3-Core
+  - ln -nfs .Build/vendor/typo3/cms/typo3 typo3
```

Comment:
Do we really need this ? I believe it should work without linking the typo3 directory.